### ISSUE TITLE
Implicit Namespace Package import unknown

### ISSUE THREAD

**Issue by Gjgarr** [microsoft/pyright#4644]
**Describe the bug**
Importing a local implicit namespace package only works when importing from an `__init__.py`

**To Reproduce**

File Structure
```
$ tree some_pkg

some_pkg
├── __init__.py
└── resources
    └── hello.txt
```

Works:
```py
# some_pkg/__init__.py - v1
from . import resources

reveal_type(resources)
# Type of "resources" is "Module("some_pkg.resources")"
```

Doesn't work (not really important to me but figured it was strange):
```py
# some_pkg/__init__.py - v2
from . import resources as resources_aliased
# error: "resources" is unknown import symbol (reportGeneralTypeIssues)

reveal_type(resources)
# Type of "resources_alias" is "Unknown"
```

Both of these cases can run successfully despite the pyright warning in v2.

If you change the `__init__.py` file to `some_mod.py`, pyright complains about both cases, despite still successfully running.

If an empty `some_pkg/resources/__init__.py` file is created, pyright recognizes the import (in all cases) and does not report the error.

**Expected behavior**
I would expect pyright to recognize `from . import resources` from any module directly under the `some_pkg` package without needing to touch `some_pkg/resources/__init__.py`

**VS Code extension or command-line**
pyright 1.1.294

**Comment by erictraut** [comment]
This is intended behavior. In your `v2`, the import statement doesn't explicitly add a symbol `resources` to the namespace of `__init__.py`. It does happen to add it to the module's namespace as an implicit side effect of the Python loader's caching mechanism. Relying on implicit side effects like this is fragile, not recommended, and often a sign of a bug. We therefore made the explicit decision not to model most of the implicit loader side effects in pyright and treat them as bugs. There are a few such side effects that we do model because they are commonly used in popular libraries, but I'd be reluctant to expand that list.
